src/main/java/com/google/sps/server/AddSchedule.java

```diff
@@ -0,0 +1,198 @@
+    try {
+      // get current date, so can add the courses to the appropriate days of the current month 
+      java.util.Calendar calendar = java.util.Calendar.getInstance();
+      int nextSun = getDateOfNextSunday(calendar);
+      int month = getCurrentMonth(calendar);
+      int year = getCurrentYear(calendar);
+
+      // for each course in the schedule, add an event to the calendar
+      Iterator<Course> courseIterator = schedule.getCourses().iterator();
+      while (courseIterator.hasNext()) {
+        Course currentCourse = courseIterator.next();
+        List<TimeRange> sectionTimes = currentCourse.getSections().get(0).getMeetingTimes();
+        for (int i = 0; i < sectionTimes.size(); i++) {
+          String startTime = calculateStartTime(sectionTimes.get(i), month, year, nextSun);
+          String endTime = calculateEndTime(sectionTimes.get(i), month, year, nextSun);
+          addEvent(currentCourse, client, calendarId, startTime, endTime);
+        }
+      }       
+    } catch (IOException e) {
+      System.out.println("Couldn't iterate through the courses in the schedule.");
+    }
+  }
+
+   /**
+   * Helper function to calculate the day of the month of the next Sunday,
+   * to know when to start the schedule. 
+   *
+   * @param calendar an instance of Calendar, so can know the current date
+   * 
+   * @return an int representing the date of next Sunday
+   */
+  public static int getDateOfNextSunday(java.util.Calendar calendar) {
+    int currentDay = calendar.get(java.util.Calendar.DAY_OF_MONTH);
+    int currentDayOfWeek = calendar.get(java.util.Calendar.DAY_OF_WEEK);
+    int nextSun = currentDay + 8 - currentDayOfWeek;
+    return nextSun;
+  }
+ 
+   /**
+   * Helper function to know the current month. 
+   *
+   * @param calendar an instance of Calendar, so can know the current date
+   * 
+   * @return an int representing the month, adjusted for DateTime's formatting
+   */
+  public static int getCurrentMonth(java.util.Calendar calendar) {
+    return calendar.get(java.util.Calendar.MONTH) + 1;
+  }
+  
+   /**
+   * Helper function to know the current year. 
+   *
+   * @param calendar an instance of Calendar, so can know the current date
+   * 
+   * @return an int representing the current year
+   */
+  public static int getCurrentYear(java.util.Calendar calendar) {
+    return calendar.get(java.util.Calendar.YEAR);
+  }
+  
+   /**
+   * Helper function to format the relevant times (start or end) for a course 
+   * being converted into a calendar event. 
+   *
+   * @param timeInMin the time, in minutes of the week, for an event (either the start or end time)
+   * @param month the current month
+   * @param year the current year
+   * @param nextSun the day of the month of the next Sunday
+   * 
+   * @return a String formatted for DateTime, to be used in creating a calendar event 
+   */
+  public static String 
+     DateTime(int timeInMin, int month, int year, int nextSun) {
```

Comment:
Is this supposed to be your `calculateDateTime` method?

Reply:
Yes, in my code it says calculateDateTime, not sure what happened but it should be fixed when I commit the changes for this review. My bad!